src/python/pants/bin/daemon_pants_runner.py

```diff
@@ -237,6 +237,27 @@ def nailgunned_stdio(cls, sock, env, handle_stdin=True):
       ) as finalizer:
         yield finalizer
 
+  @contextmanager
+  def daemon_exiter(cls, exiter):
+    """A contextmanager which temporarily overrides the exiter."""
+    try:
+      cls._exiter = exiter
+      yield exiter
+    finally:
+      pass
+
+  @contextmanager
+  def set_exiter(cls, exiter):
+    """A contextmanager which temporarily overrides the exiter."""
+    try:
+      previous_exiter = cls._exiter
+      cls._exiter = exiter
```

Comment:
It's possible that `cls._exiter` might no longer necessary, since we should always be able to use the global mutable exiter? If that's the case, then both `DaemonPantsRunner.set_exiter` and `LocalPantsRunner.set_exiter` could be merged into the same method somewhere. Maybe `with ExceptionSink.exiter_as(mk_exiter=lambda previous_exit: ..):` or something (where the `mk_exiter` function could optionally wrap the existing exiter, or replace it)?

Reply:
yes, absolutely, was thinking the same that its duplicated for no good reason - will look into this tomorrow